src/cljs/main/broadfcui/page/workspace/summary/catalog/wizard.cljs

```diff
@@ -118,27 +166,53 @@
-             (swap! state update :page-num inc)
-             (react/call :submit this))))))
+       (let [{:keys [working-attributes pages-seen required-attributes page-num]} @state
+             attributes-from-page (react/call :get-attributes (@refs "wizard-page"))
+             all-attributes (merge working-attributes attributes-from-page)
+             invalid-attributes (atom #{})]
+         (swap! state assoc :working-attributes all-attributes)
+         (swap! locals update :page-attributes assoc page-num attributes-from-page)
+         (doseq [page (conj pages-seen page-num)]
+              (let [questions (first (get-questions-for-page all-attributes (:library-schema props) page))
+                    {:keys [invalid]} (library-utils/validate-required
+                                              (convert-empty-strings all-attributes)
+                                              questions
+                                              required-attributes)]
+                (reset! invalid-attributes (clojure.set/union invalid @invalid-attributes))))
+         (swap! state assoc :invalid-properties @invalid-attributes)
+       (after-update (fn [_]
+                       (let [next-page (react/call :find-next-page this)]
+                         (swap! state #(-> %
+                                           (update :pages-seen conj page-num)
+                                           (assoc :page-num next-page)))))))))
+   :find-next-page
+   (fn [{:keys [props state]}]
+     (let [{:keys [library-schema]} props
+           {:keys [page-num working-attributes]} @state
+           next (atom (inc page-num))]
+       (while (and (not (get-questions-for-page working-attributes library-schema @next))
+                   (< @next (count (:wizard library-schema))))
+         (swap! next inc))
+       @next))
    :submit
    (fn [{:keys [props state locals]}]
-     (swap! state assoc :submitting? true :submit-error nil)
-     (endpoints/call-ajax-orch
-      {:endpoint (endpoints/save-library-metadata (:workspace-id props))
-       :payload (apply merge (:version-attributes @state) (:page-attributes @locals))
-       :headers utils/content-type=json
-       :on-done (fn [{:keys [success? get-parsed-response]}]
-                  (swap! state dissoc :submitting?)
-                  (if success?
-                    (do (modal/pop-modal)
-                        ((:request-refresh props)))
-                    (swap! state assoc :submit-error (get-parsed-response false))))}))})
+     (if (not-empty (:invalid-properties @state))
+       (swap! state assoc :validation-error "You will need to complete all required metadata attributes to be able to publish the workspace in the Data Library")
+       (let [attributes-seen (atom {})]
+         (doseq [page (:pages-seen @state)]
+           (let [attrs (nth (:page-attributes @locals) page)]
+             (swap! attributes-seen merge attrs)))
```

Comment:
You can do this with a `replace`:
```clojure
(def page-props {0 {:a 1 :b 2} 1 {:c 3 :d 4} 2 {:e 5 :f 6}})
(def pages '(0 2))
(replace page-props pages)
;; => ({:a 1 :b 2} {:e 5 :f 6})
(apply merge *1)
;; => {:a 1 :b 2 :e 5 :f 6}
```
I think the other `doseq` can be done the same way.